modules/ncc-spoke-ra/main.tf

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  spoke_vms = [
+    for ras in var.ras : {
+      ip      = ras.ip
+      vm      = ras.vm
+      vm_name = element(split("/", ras.vm), length(split("/", ras.vm)) - 1)
+    }
+  ]
+}
+
+resource "google_network_connectivity_hub" "hub" {
+  count       = var.hub.create ? 1 : 0
+  project     = var.project_id
+  name        = var.hub.name
+  description = var.hub.description
+}
+
+resource "google_network_connectivity_spoke" "spoke-ra" {
+  project  = var.project_id
+  hub      = try(google_network_connectivity_hub.hub[0].name, var.hub.name)
+  location = var.region
+  name     = var.name
+  linked_router_appliance_instances {
+    dynamic "instances" {
+      for_each = var.ras
+      content {
+        virtual_machine = instances.value["vm"]
+        ip_address      = instances.value["ip"]
+      }
+    }
+    site_to_site_data_transfer = var.data_transfer
+  }
+}
+
+resource "google_compute_router" "cr" {
```

Comment:
our convention (which we don't always follow TBH) is to use `default` as resource name when there's a single one of its type in the module

Reply:
Maybe convenient, and for sure not always followed, but I dislike that convention the same you prefer meaningful names for vars (which I agree)

Reply:
lol, feel free to ignore this :)